sn_networking/src/event.rs

```diff
@@ -232,17 +232,14 @@ impl SwarmDriver {
                                     .listen_addrs
                                     .into_iter()
                                     .filter(multiaddr_is_global)
+                                    // Strip the `/p2p/...` part of the multiaddresses
+                                    .map(|addr| multiaddr_strip_p2p(&addr))
+                                    // And deduplicate the list
+                                    .unique()
                                     .collect(),
```

Comment:
By moving these, the addresses will not be stripped in case of `local` mode (L229).